Desktop/components/JASP/Widgets/CustomMenu.qml

```diff
@@ -94,170 +94,203 @@ FocusScope
 		z		: menuShadow.z + 1
 		color	: jaspTheme.fileMenuColorBackground
 		focus	: true
+		width	: column.maxWidth + (itemScrollbar.visible ? itemScrollbar.width : 0)
+		height	: menuOffset.y + column.maxHeight > menuMaxPos.y ? menuMaxPos.y - menuOffset.y : column.maxHeight
 
 		MouseArea
 		{
 			anchors.fill	: parent
 		}
 
-	}
-
-	Column
-	{
-		id		: column
-		x		: menuRectangle.x
-		y		: menuRectangle.y + (jaspTheme.menuPadding / 2)
-		z		: menuRectangle.z + 1
-		spacing	: jaspTheme.menuSpacing
-
-		Repeater
+		JASPControl.JASPScrollBar
 		{
-			id		: repeater
-			model	: menu.props === undefined ? undefined : menu.props["model"]
+			id				: itemScrollbar
+			flickable		: itemFlickable
+			manualAnchor	: true
+			vertical		: true
+			z				: 1337
 
-			onItemAdded:
+			anchors
 			{
-				if (index === 0)
-				{
-					menuRectangle.width  = 0;
-					menuRectangle.height = 0;
-				}
+				top			: parent.top
+				right		: parent.right
+				bottom		: parent.bottom
+				margins		: 2
+			}
+		}
 
-				menuRectangle.width   = Math.max(item.width, menuRectangle.width);
-				menuRectangle.height += (item.height + jaspTheme.menuSpacing)
+		Flickable
+		{
+			id						: itemFlickable
+			anchors.fill			: parent
+			anchors.leftMargin		: jaspTheme.menuPadding / 2
```

Comment:
You should add a topMargin, because it looks a bit weird now.
See:
![image](https://user-images.githubusercontent.com/29062713/106161668-926c9180-6187-11eb-9d0b-06424645c2ef.png)